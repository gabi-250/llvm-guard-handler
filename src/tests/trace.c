#include <stdlib.h>
#include <stdio.h>

int more_indirection()
{
    return 3;
}

int get_number(int level)
{
    double dbl = 2.54645;
    if (level < 2) {
        printf("Call %d\n", level);
        return get_number(level + 1);
    } else {
        char one = '1';
        char two = 2 + '0';
        long a_long = 249238493223;
        int x = more_indirection();
        printf("dbl = %lf\n", dbl);
        printf("one = %c\n", one);
        printf("two = %c\n", two);
        printf("a long = %ld\n", a_long);
        printf("x = %d\n", x);
        return x;
    }
}

void trace()
{
    char four = '4';
    int y = 155;
    double k = 8.2345;
    int x = get_number(0);
    printf("x = %d\n", x);
    printf("y = %d\n", y);
    printf("four = %c\n", four);
    printf("k = %lf\n", k);
}

int main(int argc, char **argv)
{
    trace();
    return 0;
}
