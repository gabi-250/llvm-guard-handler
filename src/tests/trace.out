Call 0
Call 1
dbl = 2.546450
one = 1
two = 2
a long = 249238493223
x = 3
x = 3
y = 155
four = 4
k = 8.234500

